### ISSUE TITLE
Documenting the code with Haddock

### ISSUE THREAD

**Issue by ChrisEPhifer** [edwisdom/chords#11]
Since we're at a reasonably stable place with most of the core functionality, it might be time to step back and tackle documenting all of the modules using [Haddock](https://www.haskell.org/haddock/), a tool that ships with every standard Haskell installation to generate nicely formatted documentation using special comment syntax. It's very intuitive, and the results are styled precisely to ship a package to Hackage.

We can use this issue to track what modules still need documentation. Here's a checklist we can tick off as we complete documentation; this issue should also be updated as any new features are merged into `develop`:

- [x] ChordSubs.hs
- [x] Diatonic.hs
- [ ] Scale.hs

### Base

- [x] CQuality.hs
- [x] ChordSymbol.hs
- [x] Heliotonic.hs
- [x] IQuality.hs
- [x] Interval.hs
- [x] PitchClass.hs

#### Base.Chord

- [x] Chord.hs
- [x] ChordSymbol.hs
- [x] ChordShape.hs
- [x] Extension.hs
- [x] HighestNatural.hs
- [x] RawChord.hs
- [x] Root.hs
- [x] Sus.hs

#### Base.Core

- [x] Accidental.hs
- [x] Note.hs

##### Base.Core.Quality

- [x] CQuality.hs
- [x] IQuality.hs

### Common

- [x] Utils.hs

### Language

- [x] Parser.hs

**Comment by edwisdom** [comment]
@ChrisEPhifer Let me know if you think it makes sense to do something else first, but if not, I think I'll start on this issue since I think the process of documenting the functions will highlight places where we might want to have partial functions and error-handling.

**Comment by ChrisEPhifer** [comment]
I'd like to try and get my `Interval` improvement branch merged in first, I think, since it's the first place there was an obvious need to transition to functions that might error (and I'm nearly done making that change / propagating it through to the `Scale` module and `Lib`. I don't think it would hurt to get a branch going, though, that you can merge `develop` into as you work / it updates with new modules and such.

So TL;DR: Feel free to get a branch off of develop going, but be prepared to pull changes to develop into it as they come in since you'll likely want to keep these documentation changes consistent with the most up-to-date versions of everything.

**Comment by edwisdom** [comment]
Ah yes, that makes sense. I can just start with documenting the lower-level stuff once the ChordSubs branch is merged and leave the Intervals and stuff downstream from there, and hopefully we'll meet in the middle.

**Comment by ChrisEPhifer** [comment]
Makes sense! In order to test out the documentation as you're working on it, by the way, I believe you can just run

```
stack haddock
```

And this will dump HTML renders of the documentation... somewhere :)

**Comment by ChrisEPhifer** [comment]
@edwisdom if you want to sync up whatever branch you end up making for this, just checkout `develop`, pull, checkout the branch for documentation, and run `git merge develop`. That way all of the things that currently need to be documented can live in that new branch and be accurate for consistency with documentation.

Frankly, though, any new modules that aren't yet merged into `develop` should have their documentation added before their  merge is done (e.g. #17) - this way, we won't constantly need to be merging `develop` back into the branch so everything stays in sync and branches don't need to be resurrected/replicated for later documentation.

TL;DR - The documentation for a module should be among the commit history for that module's branch, and a PR for a branch without adequate documentation should be kept open until the documentation is sufficient.

**Comment by edwisdom** [comment]
I've gotten through most of these modules, but I'm leaving the ones affected by #19 for now.

**Comment by ChrisEPhifer** [comment]
I think it's safe to close this since all but one module is taken care of at this point.
